src/irmin-pack/unix/mapping_file.ml

```diff
@@ -365,95 +361,52 @@ module Make (Errs : Io_errors.S with module Io = Io.Unix) = struct
+      Util.output_int_ne file2 len;
+      ()
     in
     let* () =
       Errs.catch (fun () ->
           calculate_extents_oc ~src_is_sorted:() ~src:file1.arr ~register_entry)
     in
-    let* () = Ao.flush file2 in
-    let* () = Ao.close file2 in
+    let _ = Stdlib.flush file2 in
+    let _ = Stdlib.close_out file2 in
 
     (* Close and unlink [file1] *)
     Int_mmap.close file1;
     Io.Unix.unlink path1 |> ignore;
 
     Ok ()
 
-  let iter io f =
-    let buffer = Bytes.create (16 * 1000) in
-    let buffer_off = ref 0 in
-
-    let open Int63.Syntax in
-    let open Int63 in
-    let min a b = if a < b then a else b in
-    let entry_bytes = of_int 16 in
-    let max_entries_per_batch = of_int 1000 in
-
-    (* let max_bytes_per_batch = max_entries_per_batch * entry_bytes in *)
-    let open Result_syntax in
-    let* byte_count = Io.Unix.read_size io in
-    let entry_count = byte_count / entry_bytes in
-    (* Fmt.epr "\nbyte_count:%#d entry_count:%#d\n%!" (to_int byte_count) (to_int entry_count); *)
-    let* () =
-      if entry_count * entry_bytes <> byte_count then
-        Error (`Corrupted_mapping_file "unexpected file size")
-      else Ok ()
-    in
-
-    let rec load_batch i last_yielded_end_offset =
-      let entries_left = entry_count - i in
-      (* Fmt.epr "load_batch i:%#d, entries_left:%#d \n%!" (to_int i) (to_int entries_left); *)
-      if entries_left = zero then ()
-      else
-        let entries_in_batch = min entries_left max_entries_per_batch in
-        let off = i * entry_bytes in
-        let len = to_int (entries_in_batch * entry_bytes) in
-        Io.Unix.read_exn io ~off ~len buffer;
-        buffer_off := 0;
-        yield_entries i (i + entries_in_batch) last_yielded_end_offset
-    and yield_entries i end_i last_yielded_end_offset =
-      if i = end_i then load_batch i last_yielded_end_offset
-      else
-        let off, len =
-          (* Decoding a pair of int can't fail *)
-          (* Bytes.unsafe_to_string usage: possibly safe TODO justify safety, or convert
-             to Bytes.to_string *)
-          decode_bin_pair (Bytes.unsafe_to_string buffer) buffer_off
-        in
-        let () =
-          if off < last_yielded_end_offset then
-            let msg =
-              Fmt.str "Found off:%a len:%a but the previous entry ends as at %a"
-                Int63.pp off Int63.pp len Int63.pp last_yielded_end_offset
-            in
-            raise (Errors.Pack_error (`Corrupted_mapping_file msg))
-        in
-        f ~off ~len:(to_int len);
-        let last_yielded_end_offset = off + len in
-        yield_entries (succ i) end_i last_yielded_end_offset
-    in
-    Errs.catch (fun () -> load_batch zero zero)
+  let load_mapping_as_mmap path =
+    let mmap = Int_mmap.open_ ~fn:path ~sz:(-1) in
+    let arr = mmap.arr in
+    (* we expect the arr to hold (off,len) pairs, so size should be even *)
+    assert (BigArr1.dim arr mod 2 = 0);
+    Int_mmap.close mmap;
```

Comment:
Does this line imply that it is still possible to read `arr` while `mmap.fd` has been closed?

Reply:
Yes, that is possible.

Reply:
Thanks